packages/router/src/history.js

```diff
@@ -6,31 +6,26 @@ import { createBrowserHistory } from 'history';
 /**
  * WordPress dependencies
  */
-import { addQueryArgs, getQueryArgs, removeQueryArgs } from '@wordpress/url';
+import { buildQueryString } from '@wordpress/url';
 
 const history = createBrowserHistory();
 
 const originalHistoryPush = history.push;
 const originalHistoryReplace = history.replace;
 
+function buildSearch( params ) {
+	const queryString = buildQueryString( params );
+	return queryString.length > 0 ? '?' + queryString : queryString;
+}
+
 function push( params, state ) {
```

Comment:
Would have been nice if these had tests.

Reply:
Unfortunately this package works on the `window.history` object, it's not very unit-testable.